pkg/epp/scheduling/config.go

```diff
@@ -52,3 +59,58 @@ var defaultConfig = &SchedulerConfig{
 	picker:              defPlugin,
 	postSchedulePlugins: []plugins.PostSchedule{},
 }
+
+func CreateConfig(opts ...ConfigOption) *SchedulerConfig {
+	config := &SchedulerConfig{
+		preSchedulePlugins:  []plugins.PreSchedule{},
+		postSchedulePlugins: []plugins.PostSchedule{},
+		scorers:             map[plugins.Scorer]int{},
+		filters:             []plugins.Filter{&sheddableRequestFilterV2{}},
+		picker:              &picker.MaxScorePicker{},
+	}
+	for _, opt := range opts {
+		opt(config)
+	}
+	return config
+}
+
+type ConfigOption func(*SchedulerConfig)
+
+func WithPrefixPlugin(prefixConfig prefix.Config) ConfigOption {
+	return func(cfg *SchedulerConfig) {
+		prefixPlugin := prefix.New(prefixConfig)
+		cfg.preSchedulePlugins = append(cfg.preSchedulePlugins, prefixPlugin)
+		cfg.postSchedulePlugins = append(cfg.postSchedulePlugins, prefixPlugin)
+		cfg.scorers[prefixPlugin] = prefixConfig.Weight
+	}
+}
+
+func WithQueuePlugin(queueConfig scorer.QueueScorerConfig) ConfigOption {
+	return func(cfg *SchedulerConfig) {
+		queuePlugin := &scorer.QueueScorer{}
+		cfg.scorers[queuePlugin] = queueConfig.Weight
+	}
+}
+
+func WithKVCachePlugin(kvCacheConfig scorer.KVCacheScorerConfig) ConfigOption {
+	return func(cfg *SchedulerConfig) {
+		kvCachePlugin := &scorer.KVCacheScorer{}
+		cfg.scorers[kvCachePlugin] = kvCacheConfig.Weight
+	}
+}
```

Comment:
I have some doubts about this part.
the whole idea of this was to keep this general purpose scheduler config file while allowing specific configuration directly from `main.go`. this makes the config too specific to the plugins that exist in the repo today.
I don't think any of this is needed.

Reply:
I've added to main a code change suggestion for how to pick plugins from core alongside custom plugins. 
from my point of view it doesn't make sense to have these helper functions which their only role is to set plugins in the scheduler. we defined a general purpose `SchedulerConfig` and the use of `SchedulerWithConfig` exactly for this.

Reply:
Pls take a look at the update, I incorporated your suggestion but kept the AddPrefixPlugin helper because it has multiple extension points and you need to make sure they are all registered properly. Ideally we could check if a plugin implements a plugin extension dynamically and then add it (I believe that's how kube scheduler does it). But for now I would not add that complexity and just have that helper method.